metals/src/main/scala/scala/meta/internal/metals/BloopServers.scala

```diff
@@ -412,7 +412,13 @@ final class BloopServers(
       case Some(bloopPath) if !bloopPath.exists =>
         // we want to use the same java version as Metals, so it's ok to use java.home
         val metalsJavaHome =
-          userConfiguration.javaHome.orElse(sys.props.get("java.home"))
+          userConfiguration.javaHome
+            .orElse(sys.env.get("JAVA_HOME"))
+            .orElse(sys.props.get("java.home"))
+            // We can't cache java for nix/guix, since the path might change
+            .filterNot(path =>
+              path.contains("/nix/store") || path.contains("/gnu/store/")
```

Comment:
Maybe better to use `startsWith`? Also one is with `/` and another without :)

Reply:
Sure, changed!